packages/features/bookings/lib/handleNewBooking/createBooking.ts

```diff
@@ -0,0 +1,267 @@
+    bookerEmail,
+    rescheduleReason,
+    eventType,
+    eventTypeId,
+    reqBodyUser,
+  });
+
+  return await saveBooking(newBookingData, originalRescheduledBooking, paymentAppData, organizerUser);
+}
+
+async function saveBooking(
+  newBookingData: Prisma.BookingCreateInput,
+  originalRescheduledBooking: OriginalRescheduledBooking,
+  paymentAppData: PaymentAppData,
+  organizerUser: CreateBookingParams["organizerUser"]
+) {
+  const createBookingObj = {
+    include: {
+      user: {
+        select: { email: true, name: true, timeZone: true, username: true },
+      },
+      attendees: true,
+      payment: true,
+      references: true,
+    },
+    data: newBookingData,
+  };
+
+  if (originalRescheduledBooking?.paid && originalRescheduledBooking?.payment) {
+    const bookingPayment = originalRescheduledBooking.payment.find((payment) => payment.success);
+    if (bookingPayment) {
+      createBookingObj.data.payment = { connect: { id: bookingPayment.id } };
+    }
+  }
+
+  if (typeof paymentAppData.price === "number" && paymentAppData.price > 0) {
+    await prisma.credential.findFirstOrThrow({
+      where: {
+        appId: paymentAppData.appId,
+        ...(paymentAppData.credentialId ? { id: paymentAppData.credentialId } : { userId: organizerUser.id }),
+      },
+      select: { id: true },
+    });
+  }
+
+  return prisma.booking.create(createBookingObj);
+}
+
+function getEventTypeRel(eventTypeId: EventTypeId) {
+  return eventTypeId ? { connect: { id: eventTypeId } } : {};
+}
+
+function getAttendeesData(evt: Pick<CalendarEvent, "attendees" | "team">) {
+  //if attendee is team member, it should fetch their locale not booker's locale
+  //perhaps make email fetch request to see if his locale is stored, else
+  const attendees = evt.attendees.map((attendee) => ({
+    name: attendee.name,
+    email: attendee.email,
+    timeZone: attendee.timeZone,
+    locale: attendee.language.locale,
+  }));
+
+  if (evt.team?.members) {
+    attendees.push(
+      ...evt.team.members.map((member) => ({
+        email: member.email,
+        name: member.name,
+        timeZone: member.timeZone,
+        locale: member.language.locale,
+      }))
+    );
+  }
+
+  return attendees;
+}
+
+function buildNewBookingData(params: {
+  uid: short.SUUID;
+  evt: CalendarEvent;
+  responses: ReqBodyWithEnd["responses"] | null;
+  isConfirmedByDefault: IsConfirmedByDefault;
+  reqBodyMetadata: ReqBodyWithEnd["metadata"];
+  smsReminderNumber: AwaitedBookingData["smsReminderNumber"];
+  eventTypeSlug: AwaitedBookingData["eventTypeSlug"];
+  organizerUser: CreateBookingParams["organizerUser"];
+  reqBodyRecurringEventId: ReqBodyWithEnd["recurringEventId"];
+  originalRescheduledBooking: OriginalRescheduledBooking | null;
+  bookerEmail: AwaitedBookingData["email"];
+  rescheduleReason: AwaitedBookingData["rescheduleReason"];
+  eventType: NewBookingEventType;
+  eventTypeId: EventTypeId;
+  reqBodyUser: ReqBodyWithEnd["user"];
```

Comment:
Possible Followup: I think we could structure it better. Data that goes directly to booking can be in `booking` prop and other things could be separate props like `reqBodyUser`, `isConfirmedByDefault` etc

Trying to minimize the props here and make this function more readable.